engine/src/test/java/io/camunda/zeebe/engine/processing/incident/OutputMappingIncidentTest.java

```diff
@@ -0,0 +1,203 @@
+      ENGINE.job().withType("type").ofInstance(processInstanceKey).complete();
+    }
+
+    final var incidentRecord =
+        RecordingExporter.incidentRecords()
+            .withProcessInstanceKey(processInstanceKey)
+            .withIntent(CREATED)
+            .getFirst();
+
+    // when
+    ENGINE.variables().ofScope(processInstanceKey).withDocument(Maps.of(entry("foo", 1))).update();
+    final Record<IncidentRecordValue> resolvedIncidentRecord =
+        ENGINE.incident().ofInstance(processInstanceKey).withKey(incidentRecord.getKey()).resolve();
+
+    // then
+    assertThat(resolvedIncidentRecord.getKey()).isEqualTo(incidentRecord.getKey());
+    Assertions.assertThat(resolvedIncidentRecord.getValue())
+        .hasErrorType(ErrorType.IO_MAPPING_ERROR)
+        .hasBpmnProcessId(PROCESS_ID)
+        .hasProcessInstanceKey(processInstanceKey)
+        .hasElementId(elementId);
+    assertThat(resolvedIncidentRecord.getValue().getErrorMessage())
+        .contains("no variable found for name 'foo'");
+    assertTrue(
+        RecordingExporter.processInstanceRecords()
+            .withProcessInstanceKey(processInstanceKey)
+            .withIntent(ProcessInstanceIntent.ELEMENT_COMPLETED)
+            .exists());
+    final Map<String, String> variables = ProcessInstances.getCurrentVariables(processInstanceKey);
+    assertThat(variables).contains(entry("foo", "1"));
+    assertThat(variables).contains(entry("bar", "1"));
+  }
+
+  @Test
+  public void shouldResolveIncidentIfProcessCancelled() {
+    ENGINE.deployment().withXmlResource(bpmnModelInstance).deploy();
+
+    final long processInstanceKey = ENGINE.processInstance().ofBpmnProcessId(PROCESS_ID).create();
+
+    if (createsJob) {
+      ENGINE.job().withType("type").ofInstance(processInstanceKey).complete();
+    }
+
+    final var incidentRecord =
+        RecordingExporter.incidentRecords()
+            .withProcessInstanceKey(processInstanceKey)
+            .withIntent(CREATED)
+            .getFirst();
+
+    // when
+    ENGINE.processInstance().withInstanceKey(processInstanceKey).cancel();
+
+    // then
+    final var resolvedIncidentRecord =
+        RecordingExporter.incidentRecords()
+            .withProcessInstanceKey(processInstanceKey)
+            .withIntent(RESOLVED)
+            .getFirst();
+
+    assertThat(incidentRecord.getKey()).isEqualTo(resolvedIncidentRecord.getKey());
+    Assertions.assertThat(resolvedIncidentRecord.getValue())
+        .hasErrorType(ErrorType.IO_MAPPING_ERROR)
+        .hasBpmnProcessId(PROCESS_ID)
+        .hasProcessInstanceKey(processInstanceKey)
+        .hasElementId(elementId);
+    assertThat(resolvedIncidentRecord.getValue().getErrorMessage())
+        .contains("no variable found for name 'foo'");
+    assertTrue(
+        RecordingExporter.processInstanceRecords()
+            .withProcessInstanceKey(processInstanceKey)
+            .withIntent(ProcessInstanceIntent.ELEMENT_TERMINATED)
+            .exists());
```

Comment:
:x: Same as above, we need to restrict the query to the process instance (i.e. the process).